### ISSUE TITLE
Filter "text" area is narrow and won't expand

### ISSUE THREAD

**Issue by johndoe1969** [RealRaven2000/FiltaQuilla#145]
I was using CentOS 8 and Thunderbird 78. I installed Fedora 35 and Thunderbird 91.4.0 64-bit and transferred my Thunderbird profile into new setup.

When I edit an existing filter or add a new one. The filter "text" area is very narrow when selecting one of the FiltaQuilla options such as (Subject|Header|Body) Regex Match, BCC, or Folder Name.

In the pics below...
One I just choose Subject for the match and the filter text area is normal/wide. The other pic I choose Subject Regex Match and you can see how narrow the filter text area is. You can't drag it to make it wider.

I've uninstalled and reinstalled version 3.2 of FiltaQuilla. I've disabled/enabled FiltaQuilla and I've restarted Thunderbird numerous times. No change.

![Screenshot1](https://user-images.githubusercontent.com/50086425/149644351-ead8b53a-13cf-4053-bf11-8add962c5460.png)
![Screenshot2](https://user-images.githubusercontent.com/50086425/149644352-8f26f4bc-81ff-4a9d-b313-11491f5a4365.png)

Any ideas?

**Comment by RealRaven2000** [comment]
Looks like the vendor specific css rule -moz-box-flex isn't working anymore, so I replaced it with flex-grow:

```css
.ruleactiontarget[type="filtaquilla@mesquilla.com#launchFile"],
.ruleactiontarget[type="filtaquilla@mesquilla.com#runFile"],
.ruleactiontarget[type="filtaquilla@mesquilla.com#addSender"],
.ruleactiontarget[type="filtaquilla@mesquilla.com#saveAttachment"] ,
.ruleactiontarget[type="filtaquilla@mesquilla.com#detachAttachments"] ,
.ruleactiontarget[type="filtaquilla@mesquilla.com#javascriptAction"] ,
.ruleactiontarget[type="filtaquilla@mesquilla.com#javascriptActionBody"] ,
.ruleactiontarget[type="filtaquilla@mesquilla.com#saveMessageAsFile"] {
  flex-grow:  2;
  /* -moz-box-flex: 1;  */
}
```

I make the grow value 2 so  it griws more than the selector on the left. Here is the updated version:

[filtaquilla-3.2.1pre15.zip](https://github.com/RealRaven2000/FiltaQuilla/files/7877190/filtaquilla-3.2.1pre15.zip)

To install, download the zip file and (without unpacking) drag into Thunderbird Add-ons Manager.

**Comment by johndoe1969** [comment]
See pic below. On my system, after removing 3.2 and installing 3.2.1pre15,  it "expanded" all fields and made the filter "text" area disappear.

![Screenshot](https://user-images.githubusercontent.com/50086425/149665984-c287aeff-0d85-47c6-aad2-aac9543f56cf.png)

**Comment by RealRaven2000** [comment]
> See pic below. On my system, after removing 3.2 and installing 3.2.1pre15, it "expanded" all fields and made the filter "text" area disappear.
> 

What version of Thunderbird? Apart from "Subject Regex Match / matches" and "Body / contains", what other types of conditions does it have? It works for me here: on TB 91.5:

![image](https://user-images.githubusercontent.com/6314674/149667928-ebac41b5-02ff-44f7-b8cf-fd7d9b61f475.png)

**Comment by johndoe1969** [comment]
I have Tbird 91.4.

So I installed it again, I went to configuration/preferences and de-selected all Filter Actions & Search Terms. I have left all Filter Actions unchecked and I have selected (Subject|Body|Header|Subject or Body|Regex case-insensitive) Regex Match, and Folder Name.

All seems to be working now. I can't explain why I got what I got earlier when the filter field disappeared.

Thanks!

**Comment by RealRaven2000** [comment]
> All seems to be working now. I can't explain why I got what I got earlier when the filter field disappeared.
> 
> Thanks!

Very strange - maybe you should have restarted Thunderbird. I know our new mail extensions should theoretically be restartless but sometimes old code (or in this case CSS rules) may hang around. Also there is no reliable mechanism to remove old conditions from the filter editor, that's why sometimes you may see duplications in the dropdown menus after installing a new FQ version. 

I will close this issue when I release the next version of FiltaQuilla.

**Comment by RealRaven2000** [comment]
Implemented in 3.3, officially released today (21/02/22)
